server/src/test/java/org/elasticsearch/snapshots/SnapshotShardsServiceIT.java

```diff
@@ -85,13 +83,11 @@ public void testRetryPostingSnapshotStatusMessages() throws Exception {
         logger.info("--> blocking repository");
         String blockedNode = blockNodeWithIndex("repo", "test");
 
-        execute("create snapshot repo.snapshot table doc.test with (wait_for_completion = false)");
+        execute("create snapshot repo.snapshot table doc.test with (wait_for_completion = true)");
         waitForBlock(blockedNode, "repo", TimeValue.timeValueSeconds(60));
 
-        final SnapshotId snapshotId = client().admin().cluster()
-            .execute(GetSnapshotsAction.INSTANCE, new GetSnapshotsRequest("repo", new String[] { "snapshot" }))
-            .get()
-            .getSnapshots().get(0).snapshotId();
+        SnapshotInfo snapshotInfo = snapshotInfo("repo", "snapshot");
+        final SnapshotId snapshotId = snapshotInfo.snapshotId();
 
```

Comment:
This is not working as sys.snapshot returns an empty result set.

MockRepository is correctly injected in both cases (SQL and not SQL). 

Debugging shows that both methods use `BlobStoreRepository.getRepositoryData` but SQL approach returns `EMPTY` immediately and `GetSnapshotRequest` repeats the query until it sees non `EMPTY`. 

It's probably some specific for admin client logic with retries. Will put it on hold for now.